Source/WTF/wtf/PageReservation.h

```diff
@@ -89,41 +89,48 @@ class PageReservation : private PageBlock {
         return m_committed;
     }
 
-    static PageReservation reserve(size_t size, OSAllocator::Usage usage = OSAllocator::UnknownUsage, bool writable = true, bool executable = false, bool jitCageEnabled = false)
+    static PageReservation reserve(size_t size, OSAllocator::Usage usage = OSAllocator::UnknownUsage, std::optional<uintptr_t> address = { }, bool writable = true, bool executable = false, bool commit = false, bool jitCageEnabled = false)
     {
         ASSERT(isPageAligned(size));
-        return PageReservation(OSAllocator::reserveUncommitted(size, usage, writable, executable, jitCageEnabled, false), size, writable, executable, jitCageEnabled, false);
+        ASSERT(isPageAligned(address.value_or(0)));
+        if (commit)
+            return PageReservation(OSAllocator::reserveAndCommit(size, usage, (void*)address.value_or(0),             writable, executable, jitCageEnabled, false), size, writable, executable, jitCageEnabled, false);
+        return PageReservation(OSAllocator::reserveUncommitted(size, usage, (void*)address.value_or(0),             writable, executable, jitCageEnabled, false), size, writable, executable, jitCageEnabled, false);
```

Comment:
We should remove spaces between `value_or(0),` and `writable`. Also, let's use `std::bit_cast<void*>(...)`.